### PR TITLE
Feature/task menu

### PR DESCRIPTION
**Pull request by Vartanchik**
- Added three dots task menu.
- Added edit task pop-up.
- Added complete task pop-up.
- Added delete task pop-up.

### CODE REVIEW

File: src/app/shared-module/validation/number.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      }
+      if (min && value < min) {
+        return { numberLessThanMin: min };
+      }
+      if (max && value > max) {
+        return { numberMoreThanMax: max };
+      }
+      return null;
+    };
+  }
```
Comment (Sihuta) [inline src/app/shared-module/validation/number.ts]: Validators contain methods for the similar logic you've implemented here. Check for the `settingsValidators` (settings-pop-up.component) in #12. It's not that critical but the less code, the better, right?

File: src/app/modals/complete-task-pop-up/complete-task-pop-up.component.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+})
+export class CompleteTaskPopUpComponent {
+  completeTaskResult?: any;
+
+  constructor(private taskService: TaskService, private dialogRef: MatDialog) {}
+
+  onSubmit() {
+    // TODO
+    this.taskService.completeCurrentTask().subscribe({
+      next: () => {
```
Comment (BaseCampGL2023) [inline src/app/modals/complete-task-pop-up/complete-task-pop-up.component.ts:17]: .pipe(catchError(val => { ___ ; return of(___)}))

File: src/app/shared-module/validation/number.ts (hunk 1 above)
Comment (Vartanchik) [reply src/app/shared-module/validation/number.ts]: I agree with you)

File: src/app/modals/complete-task-pop-up/complete-task-pop-up.component.ts (hunk 2 above)
Comment (Vartanchik) [reply src/app/modals/complete-task-pop-up/complete-task-pop-up.component.ts:17]: As I understood we need to use pipe in the serviece to filter Observable. And if we need pass error message to the component, we get it there from subscribe?

Comment (BaseCampGL2023) [reply src/app/modals/complete-task-pop-up/complete-task-pop-up.component.ts:17]: we can still access it from the pipe, unless you have pipe(catchError) in taskService which already handled the error

**Pull request merged**